lib/extensions/ar_virtual.rb

```diff
@@ -617,10 +617,11 @@ def find_with_associations
       def select(*fields)
         return super if block_given? || fields.empty?
         # support virtual attributes by adding an alias to the sql phrase for the column
+        # it does not add an as() if the column already has an as
         # this code is based upon _select()
         fields.flatten!
         fields.map! do |field|
-          if virtual_attribute?(field) && (arel = klass.arel_attribute(field))
+          if virtual_attribute?(field) && (arel = klass.arel_attribute(field)) && arel.respond_to?(:as)
```

Comment:
No, I was suggesting that we are already overriding `.select` at this point, and so the addition of an extra conditional at this point is probably not going to make much more of a difference from what we already have here, so my comment is probably making a bigger deal out of this than it really needs to.